src/isolatable-hybrid.store.spec.ts

```diff
@@ -130,4 +133,78 @@ describe('IsolatableHybridConfigurationStore', () => {
       });
     },
   );
+
+  describe('Serving store hooks', () => {
+    let syncStore: ISyncStore<number>;
+    let persistentStore: IPairedAsyncStore<number> & {
+      // custom persistent store method to set serving store entries
+      setServingEntries(entries?: Record<string, number>): Promise<void>;
+    };
+    let store: IsolatableHybridConfigurationStore<number>;
+
+    const initialEntries = { a: 1, b: 2 };
+    const newEntries = { a: 3, b: 4 };
+
+    beforeEach(() => {
+      let dummyStorage: Record<string, number> = initialEntries;
+      let rehydrationCallback: (entries?: Record<string, number>) => Promise<void>;
+      syncStore = new MemoryStore();
+      persistentStore = {
+        async entries(): Promise<Record<string, number>> {
+          return dummyStorage;
+        },
+        async setEntries(entries: Record<string, number>): Promise<void> {
+          dummyStorage = entries;
+        },
+        async isExpired(): Promise<boolean> {
+          return false;
+        },
+        isInitialized(): boolean {
+          return true;
+        },
+        // Optional register rehydrate hook provided
+        registerRehydrate(rehydrate: (entries?: Record<string, number>) => Promise<void>): void {
+          rehydrationCallback = rehydrate;
+        },
+        // Below method custom to this persistent store
+        async setServingEntries(entries?: Record<string, number>): Promise<void> {
+          rehydrationCallback && (await rehydrationCallback(entries));
+        },
+      };
+
+      store = new IsolatableHybridConfigurationStore(syncStore, persistentStore, 'expired');
+    });
+
+    it('Provides hook to update serving store from persistent store', async () => {
+      await store.init();
+      // Expect sync store to be initialized with the async store
+      expect((await persistentStore.entries())?.['a']).toBe(initialEntries['a']);
```

Comment:
at first I used the hard-coded values but I think it actually easier to follow when deferencing `initialEntries` and `newEntries` so you don't have to remember `1` is old and `3` is new.